### PR TITLE
New docs for frame and series

### PR DESCRIPTION
**Pull request by r-brink**
Finished some work that was open from Friday

### CODE REVIEW

File: py-polars/polars/eager/frame.py [hunk 1]
```diff
@@ -1580,6 +1601,26 @@ def tail(self, length: int = 5) -> "DataFrame":
     def drop_nulls(self, subset: Optional[tp.List[str]] = None) -> "DataFrame":
         """
         Return a new DataFrame where the null values are dropped.
+
+         Examples
+        --------
+        >>> df = pl.DataFrame({
```
Comment (ritchie46) [inline py-polars/polars/eager/frame.py:1607]: Maybe show an example with a DataFrame that has null data?

File: py-polars/polars/eager/frame.py [hunk 2]
```diff
@@ -2677,6 +2815,17 @@ def row(self, index: int) -> Tuple[Any]:
...
         Convert columnar data to rows as python tuples.
+
+        Examples
+        --------
+        >>> df = pl.DataFrame({
+        >>>     "foo": [1, 2, 3],
+        >>>     "bar": [6, 7, 8],
+        >>>     "ham": ['a', 'b', 'c']
+        >>>     })
+        >>> df.row(2)
```
Comment (ritchie46) [inline py-polars/polars/eager/frame.py:2826]: I think this example should be added to `DataFrame.row`, this is `DataFrame.rows`

File: py-polars/polars/eager/series.py [hunk 3]
```diff
@@ -1377,6 +1389,18 @@ def explode(self) -> "Series":
         """
         Explode a list or utf8 Series. This means that every item is expanded to a new row.
 
+        Examples
+        --------
+        >>> s = pl.Series('1', ['x' 'y', 'z'])
```
Comment (ritchie46) [inline py-polars/polars/eager/series.py:1394]: This example is only interesting with list data. It flattens the lists.

File: py-polars/polars/eager/series.py [hunk 4]
```diff
@@ -1897,6 +1953,31 @@ def rolling_min(
...
+        shape: (5,)
+        Series: '' [i64]
+        [
+                null
+                null
+                100
+                200
+                300
+        ]
+        >>> s.rolling_min(window_size=3, min_periods=2)
```
Comment (ritchie46) [inline py-polars/polars/eager/series.py:1970]: Maybe show and example that differs from output from the first one?

File: py-polars/polars/eager/series.py [hunk 5]
```diff
@@ -1929,6 +2010,31 @@ def rolling_max(
...
+        shape: (5,)
+        Series: '' [i64]
+        [
+                null
+                null
+                300
+                400
+                500
+        ]
+        >>> s.rolling_max(window_size=3, min_periods=2))
```
Comment (ritchie46) [inline py-polars/polars/eager/series.py:2027]: Maybe show an example that differs from output from the first one?

**Review by ritchie46 — CHANGES_REQUESTED** [review]
Thanks a lot @r-brink , this truly make the library better. I have a few remarks.

File: py-polars/polars/eager/series.py (hunk 5 above)
Comment (r-brink) [reply py-polars/polars/eager/series.py]: I am going to leave second examples out. I don't find them very clear either. Will come back to it when I have played around with it and found a better example

**Pull request merged**